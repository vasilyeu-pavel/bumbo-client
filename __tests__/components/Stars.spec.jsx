/* global expect */

import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Stars from '../../app/bundles/World/components/common/Stars';

describe('<Stars /> component', () => {
  const minProps = { count: 2 };

  it('should render component with class "stars"', () => {
    const stars = renderer.create(<Stars {...minProps} />).toJSON();

    expect(stars).toMatchSnapshot();
  });

  it('should render component with class "stars--small"', () => {
    const stars = shallow(<Stars {...minProps} size='small' />);

    expect(stars.hasClass('stars--small')).toEqual(true);
  });

  it('should render component with class "stars--huge"', () => {
    const stars = shallow(<Stars {...minProps} size='huge' />);

    expect(stars.hasClass('stars--huge')).toEqual(true);
  });

  it('should render component with yellow icons', () => {
    const stars = renderer.create(<Stars {...minProps} color='yellow' />).toJSON();

    expect(stars).toMatchSnapshot();
  });

  it('should render component with orange icons', () => {
    const stars = renderer.create(<Stars {...minProps} color='orange' />).toJSON();

    expect(stars).toMatchSnapshot();
  });

  it(`should render ${minProps.count} SVG elements`, () => {
    const stars = shallow(<Stars {...minProps} />);

    expect(stars.find('svg')).toHaveLength(minProps.count);
  });
});
