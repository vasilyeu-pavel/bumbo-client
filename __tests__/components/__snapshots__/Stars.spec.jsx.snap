exports[`<Stars /> component should render component with class "stars" 1`] = `
<div
  className="stars">
  <div
    className="stars__item">
    <svg
      fill="gray"
      viewBox="0 0 32 31">
      <path
        d="M16 1.996l4.273 8.818 9.717 1.375-7.007 7.046 1.613 9.76L16 24.55l-8.6 4.448 1.613-9.76L2.01 12.19l9.717-1.376L16 1.996M16 0c-.895 0-1.71.505-2.09 1.293l-3.62 7.513-8.326 1.24c-.855.126-1.565.71-1.84 1.51-.276.8-.068 1.682.536 2.284l6.1 6.094-1.415 8.45c-.142.85.226 1.704.95 2.203.396.273.862.41 1.33.41.384 0 .77-.093 1.116-.278L16 26.933l7.26 3.784c.348.184.733.278 1.117.278.467 0 .934-.138 1.332-.41.72-.5 1.09-1.354.943-2.204l-1.415-8.45 6.102-6.094c.604-.602.81-1.483.536-2.284-.276-.8-.985-1.384-1.84-1.51l-8.326-1.24-3.62-7.513C17.712.506 16.896 0 16 0z" />
    </svg>
  </div>
  <div
    className="stars__item">
    <svg
      fill="gray"
      viewBox="0 0 32 31">
      <path
        d="M16 1.996l4.273 8.818 9.717 1.375-7.007 7.046 1.613 9.76L16 24.55l-8.6 4.448 1.613-9.76L2.01 12.19l9.717-1.376L16 1.996M16 0c-.895 0-1.71.505-2.09 1.293l-3.62 7.513-8.326 1.24c-.855.126-1.565.71-1.84 1.51-.276.8-.068 1.682.536 2.284l6.1 6.094-1.415 8.45c-.142.85.226 1.704.95 2.203.396.273.862.41 1.33.41.384 0 .77-.093 1.116-.278L16 26.933l7.26 3.784c.348.184.733.278 1.117.278.467 0 .934-.138 1.332-.41.72-.5 1.09-1.354.943-2.204l-1.415-8.45 6.102-6.094c.604-.602.81-1.483.536-2.284-.276-.8-.985-1.384-1.84-1.51l-8.326-1.24-3.62-7.513C17.712.506 16.896 0 16 0z" />
    </svg>
  </div>
</div>
`;

exports[`<Stars /> component should render component with orange icons 1`] = `
<div
  className="stars">
  <div
    className="stars__item">
    <svg
      fill="#ff9c00"
      viewBox="0 0 32 31">
      <path
        d="M16 1.996l4.273 8.818 9.717 1.375-7.007 7.046 1.613 9.76L16 24.55l-8.6 4.448 1.613-9.76L2.01 12.19l9.717-1.376L16 1.996M16 0c-.895 0-1.71.505-2.09 1.293l-3.62 7.513-8.326 1.24c-.855.126-1.565.71-1.84 1.51-.276.8-.068 1.682.536 2.284l6.1 6.094-1.415 8.45c-.142.85.226 1.704.95 2.203.396.273.862.41 1.33.41.384 0 .77-.093 1.116-.278L16 26.933l7.26 3.784c.348.184.733.278 1.117.278.467 0 .934-.138 1.332-.41.72-.5 1.09-1.354.943-2.204l-1.415-8.45 6.102-6.094c.604-.602.81-1.483.536-2.284-.276-.8-.985-1.384-1.84-1.51l-8.326-1.24-3.62-7.513C17.712.506 16.896 0 16 0z" />
    </svg>
  </div>
  <div
    className="stars__item">
    <svg
      fill="#ff9c00"
      viewBox="0 0 32 31">
      <path
        d="M16 1.996l4.273 8.818 9.717 1.375-7.007 7.046 1.613 9.76L16 24.55l-8.6 4.448 1.613-9.76L2.01 12.19l9.717-1.376L16 1.996M16 0c-.895 0-1.71.505-2.09 1.293l-3.62 7.513-8.326 1.24c-.855.126-1.565.71-1.84 1.51-.276.8-.068 1.682.536 2.284l6.1 6.094-1.415 8.45c-.142.85.226 1.704.95 2.203.396.273.862.41 1.33.41.384 0 .77-.093 1.116-.278L16 26.933l7.26 3.784c.348.184.733.278 1.117.278.467 0 .934-.138 1.332-.41.72-.5 1.09-1.354.943-2.204l-1.415-8.45 6.102-6.094c.604-.602.81-1.483.536-2.284-.276-.8-.985-1.384-1.84-1.51l-8.326-1.24-3.62-7.513C17.712.506 16.896 0 16 0z" />
    </svg>
  </div>
</div>
`;

exports[`<Stars /> component should render component with yellow icons 1`] = `
<div
  className="stars">
  <div
    className="stars__item">
    <svg
      fill="#fff06e"
      viewBox="0 0 32 31">
      <path
        d="M16 1.996l4.273 8.818 9.717 1.375-7.007 7.046 1.613 9.76L16 24.55l-8.6 4.448 1.613-9.76L2.01 12.19l9.717-1.376L16 1.996M16 0c-.895 0-1.71.505-2.09 1.293l-3.62 7.513-8.326 1.24c-.855.126-1.565.71-1.84 1.51-.276.8-.068 1.682.536 2.284l6.1 6.094-1.415 8.45c-.142.85.226 1.704.95 2.203.396.273.862.41 1.33.41.384 0 .77-.093 1.116-.278L16 26.933l7.26 3.784c.348.184.733.278 1.117.278.467 0 .934-.138 1.332-.41.72-.5 1.09-1.354.943-2.204l-1.415-8.45 6.102-6.094c.604-.602.81-1.483.536-2.284-.276-.8-.985-1.384-1.84-1.51l-8.326-1.24-3.62-7.513C17.712.506 16.896 0 16 0z" />
    </svg>
  </div>
  <div
    className="stars__item">
    <svg
      fill="#fff06e"
      viewBox="0 0 32 31">
      <path
        d="M16 1.996l4.273 8.818 9.717 1.375-7.007 7.046 1.613 9.76L16 24.55l-8.6 4.448 1.613-9.76L2.01 12.19l9.717-1.376L16 1.996M16 0c-.895 0-1.71.505-2.09 1.293l-3.62 7.513-8.326 1.24c-.855.126-1.565.71-1.84 1.51-.276.8-.068 1.682.536 2.284l6.1 6.094-1.415 8.45c-.142.85.226 1.704.95 2.203.396.273.862.41 1.33.41.384 0 .77-.093 1.116-.278L16 26.933l7.26 3.784c.348.184.733.278 1.117.278.467 0 .934-.138 1.332-.41.72-.5 1.09-1.354.943-2.204l-1.415-8.45 6.102-6.094c.604-.602.81-1.483.536-2.284-.276-.8-.985-1.384-1.84-1.51l-8.326-1.24-3.62-7.513C17.712.506 16.896 0 16 0z" />
    </svg>
  </div>
</div>
`;
